sdk/src/client/api/block_builder/transaction_builder/mod.rs

```diff
@@ -688,99 +690,113 @@ impl TransactionBuilder {
+    len: usize,
+}
 
-                required_address.is_ed25519()
-            });
+impl OrderedInputs {
+    pub(crate) fn iter(&self) -> OrderedInputsIter<&Address, &InputSigningData> {
+        self.into_iter()
+    }
 
-        for input in account_nft_address_inputs {
-            let required_address = input
-                .output
-                .required_address(commitment_slot_index, committable_age_range)?
-                .expect("expiration unlockable outputs already filtered out");
+    pub(crate) fn insert(&mut self, required_address: Address, input: InputSigningData) {
+        if required_address.is_ed25519_backed() {
+            self.ed25519.push_back(input);
+        } else {
+            self.other.entry(required_address).or_default().push_back(input);
+        }
+        self.len += 1;
+    }
 
-            match sorted_inputs
-                .iter()
-                .position(|input_signing_data| match required_address {
-                    Address::Account(unlock_address) => {
-                        if let Output::Account(account_output) = &input_signing_data.output {
-                            *unlock_address.account_id()
-                                == account_output.account_id_non_null(input_signing_data.output_id())
-                        } else {
-                            false
-                        }
-                    }
-                    Address::Nft(unlock_address) => {
-                        if let Output::Nft(nft_output) = &input_signing_data.output {
-                            *unlock_address.nft_id() == nft_output.nft_id_non_null(input_signing_data.output_id())
-                        } else {
-                            false
-                        }
+    pub(crate) fn len(&self) -> usize {
+        self.len
+    }
+
+    pub(crate) fn is_empty(&self) -> bool {
+        self.len() == 0
+    }
+}
+
+#[derive(Clone, Debug)]
+pub(crate) struct OrderedInputsIter<A: Borrow<Address> + Ord + core::hash::Hash, I: Borrow<InputSigningData>> {
+    queue: VecDeque<I>,
+    other: BTreeMap<A, VecDeque<I>>,
+}
+
+impl<A: Borrow<Address> + Ord + core::hash::Hash, I: Borrow<InputSigningData>> Iterator for OrderedInputsIter<A, I> {
+    type Item = I;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        if let Some(input) = self.queue.pop_front() {
+            // Add associated inputs to the front of the queue
+            match &input.borrow().output {
+                Output::Account(account_output) => {
+                    for input in self
+                        .other
+                        .remove(&Address::Account(AccountAddress::new(
+                            account_output.account_id_non_null(input.borrow().output_id()),
+                        )))
+                        .into_iter()
+                        .flatten()
+                        .rev()
+                    {
+                        self.queue.push_front(input);
```

Comment:
How does this handle multiple account outputs owned by other account outputs?
Like there is 
Account A controlled by an Ed25519 address
Account B controlled by Account A
Account C controlled by Account B
where does it make sure that Account B is before Account C, so the unlock can reference it with a lower index?

Reply:
it's the purpose of the queue. it will get the next input, then find the inputs that depend on it and add those to the front of the queue, so it looks like this:
```
Account A controlled by an Ed25519 address
Account B controlled by Account A
Account C controlled by Account B
Account D controlled by an Ed25519 address
Account E controlled by an Ed25519 address
Account F controlled by Account D

Queue at each next call:
[A, D, E] -> A
[B, D, E] -> B
[C, D, E] -> C
[D, E] -> D
[F, E] -> F
[E] -> E

Reply:
But what if the queue is not already ordered as it needs to be (or is that done somewhere)? So like this
Account A controlled by an Ed25519 address
Account C controlled by Account B
Account B controlled by Account A

Reply:
The queue initially only contains the ed25519 controlled outputs, so that would look like this:

```
[A] -> A
[B] -> B
[C] -> C
```

As each dependent is looked up from the map on each iteration

Reply:
Ok I added debug logs to understand it better like this
```Rust
    fn next(&mut self) -> Option<Self::Item> {
        log::debug!("queue: {:?}", self.queue);
        if let Some(input) = self.queue.pop_front() {
            // Add associated inputs to the front of the queue
            match &input.borrow().output {
                Output::Account(account_output) => {
                    for input in self
                        .other
                        .remove(&Address::Account(AccountAddress::new(
                            account_output.account_id_non_null(input.borrow().output_id()),
                        )))
                        .into_iter()
                        .flatten()
                        .rev()
                    {
                        self.queue.push_front(input);
                        log::debug!("queue 2: {:?}", self.queue);
```
and this is the output (I removed the actual self.queue here to save space)
```
2024-03-13 19:32:14 (UTC) iota_sdk::client::api::block_builder::transaction_builder DEBUG Selecting input OutputId { id: "0x8d144dd04605a0d249abed652c320b580f96c7119d1296bb600eadb1cd1160155a0000000000", transaction_id: TransactionId { id: "0x8d144dd04605a0d249abed652c320b580f96c7119d1296bb600eadb1cd1160155a000000", slot_index: SlotIndex(90) }, output_index: 0 }
2024-03-13 19:32:14 (UTC) iota_sdk::client::api::block_builder::transaction_builder::transition DEBUG No transition of OutputId { id: "0x8d144dd04605a0d249abed652c320b580f96c7119d1296bb600eadb1cd1160155a0000000000", transaction_id: TransactionId { id: "0x8d144dd04605a0d249abed652c320b580f96c7119d1296bb600eadb1cd1160155a000000", slot_index: SlotIndex(90) }, output_index: 0 }/AccountId(0x2222222222222222222222222222222222222222222222222222222222222222) as output already exists
2024-03-13 19:32:14 (UTC) iota_sdk::client::api::block_builder::transaction_builder DEBUG Selecting input OutputId { id: "0x1f0cdc1bf552449906ea7139f6a29ddd0e7619915841b6de8cf73032226df55f5a0000000000", transaction_id: TransactionId { id: "0x1f0cdc1bf552449906ea7139f6a29ddd0e7619915841b6de8cf73032226df55f5a000000", slot_index: SlotIndex(90) }, output_index: 0 }
2024-03-13 19:32:14 (UTC) iota_sdk::client::api::block_builder::transaction_builder::transition DEBUG No transition of OutputId { id: "0x1f0cdc1bf552449906ea7139f6a29ddd0e7619915841b6de8cf73032226df55f5a0000000000", transaction_id: TransactionId { id: "0x1f0cdc1bf552449906ea7139f6a29ddd0e7619915841b6de8cf73032226df55f5a000000", slot_index: SlotIndex(90) }, output_index: 0 }/AccountId(0x1111111111111111111111111111111111111111111111111111111111111111) as output already exists
2024-03-13 19:32:14 (UTC) iota_sdk::client::api::block_builder::transaction_builder DEBUG Adding Account(AccountId(0x2222222222222222222222222222222222222222222222222222222222222222)) from input OutputId { id: "0x1f0cdc1bf552449906ea7139f6a29ddd0e7619915841b6de8cf73032226df55f5a0000000000", transaction_id: TransactionId { id: "0x1f0cdc1bf552449906ea7139f6a29ddd0e7619915841b6de8cf73032226df55f5a000000", slot_index: SlotIndex(90) }, output_index: 0 }
2024-03-13 19:32:14 (UTC) iota_sdk::client::api::block_builder::transaction_builder DEBUG Selecting input OutputId { id: "0xe21a070f8a2d685ea29a701f171271cce9823f3d83e00ccc9327e96ec7e63eb05a0000000000", transaction_id: TransactionId { id: "0xe21a070f8a2d685ea29a701f171271cce9823f3d83e00ccc9327e96ec7e63eb05a000000", slot_index: SlotIndex(90) }, output_index: 0 }
2024-03-13 19:32:14 (UTC) iota_sdk::client::api::block_builder::transaction_builder::transition DEBUG No transition of OutputId { id: "0xe21a070f8a2d685ea29a701f171271cce9823f3d83e00ccc9327e96ec7e63eb05a0000000000", transaction_id: TransactionId { id: "0xe21a070f8a2d685ea29a701f171271cce9823f3d83e00ccc9327e96ec7e63eb05a000000", slot_index: SlotIndex(90) }, output_index: 0 }/AccountId(0x3333333333333333333333333333333333333333333333333333333333333333) as output already exists
2024-03-13 19:32:14 (UTC) iota_sdk::client::api::block_builder::transaction_builder DEBUG Adding Account(AccountId(0x1111111111111111111111111111111111111111111111111111111111111111)) from input OutputId { id: "0xe21a070f8a2d685ea29a701f171271cce9823f3d83e00ccc9327e96ec7e63eb05a0000000000", transaction_id: TransactionId { id: "0xe21a070f8a2d685ea29a701f171271cce9823f3d83e00ccc9327e96ec7e63eb05a000000", slot_index: SlotIndex(90) }, output_index: 0 }
2024-03-13 19:32:14 (UTC) iota_sdk::client::api::block_builder::transaction_builder::requirement DEBUG Adding Account(AccountId(0x3333333333333333333333333333333333333333333333333333333333333333)) from output
2024-03-13 19:32:14 (UTC) iota_sdk::client::api::block_builder::transaction_builder::requirement DEBUG Adding Account(AccountId(0x1111111111111111111111111111111111111111111111111111111111111111)) from output
2024-03-13 19:32:14 (UTC) iota_sdk::client::api::block_builder::transaction_builder::requirement DEBUG Adding Account(AccountId(0x2222222222222222222222222222222222222222222222222222222222222222)) from output
2024-03-13 19:32:14 (UTC) iota_sdk::client::api::block_builder::transaction_builder::requirement DEBUG Fulfilling requirement Account(AccountId(0x2222222222222222222222222222222222222222222222222222222222222222))
2024-03-13 19:32:14 (UTC) iota_sdk::client::api::block_builder::transaction_builder DEBUG queue:
2024-03-13 19:32:14 (UTC) iota_sdk::client::api::block_builder::transaction_builder DEBUG queue 2
2024-03-13 19:32:14 (UTC) iota_sdk::client::api::block_builder::transaction_builder::requirement::account DEBUG AccountId(0x2222222222222222222222222222222222222222222222222222222222222222) requirement already fulfilled by OutputId { id: "0x8d144dd04605a0d249abed652c320b580f96c7119d1296bb600eadb1cd1160155a0000000000", transaction_id: TransactionId { id: "0x8d144dd04605a0d249abed652c320b580f96c7119d1296bb600eadb1cd1160155a000000", slot_index: SlotIndex(90) }, output_index: 0 }
2024-03-13 19:32:14 (UTC) iota_sdk::client::api::block_builder::transaction_builder::requirement DEBUG Fulfilling requirement Account(AccountId(0x1111111111111111111111111111111111111111111111111111111111111111))
2024-03-13 19:32:14 (UTC) iota_sdk::client::api::block_builder::transaction_builder DEBUG queue:
2024-03-13 19:32:14 (UTC) iota_sdk::client::api::block_builder::transaction_builder DEBUG queue 2
2024-03-13 19:32:14 (UTC) iota_sdk::client::api::block_builder::transaction_builder DEBUG queue:
2024-03-13 19:32:14 (UTC) iota_sdk::client::api::block_builder::transaction_builder DEBUG queue 2
2024-03-13 19:32:14 (UTC) iota_sdk::client::api::block_builder::transaction_builder::requirement::account DEBUG AccountId(0x1111111111111111111111111111111111111111111111111111111111111111) requirement already fulfilled by OutputId { id: "0x1f0cdc1bf552449906ea7139f6a29ddd0e7619915841b6de8cf73032226df55f5a0000000000", transaction_id: TransactionId { id: "0x1f0cdc1bf552449906ea7139f6a29ddd0e7619915841b6de8cf73032226df55f5a000000", slot_index: SlotIndex(90) }, output_index: 0 }
2024-03-13 19:32:14 (UTC) iota_sdk::client::api::block_builder::transaction_builder::requirement DEBUG Fulfilling requirement Account(AccountId(0x3333333333333333333333333333333333333333333333333333333333333333))
2024-03-13 19:32:14 (UTC) iota_sdk::client::api::block_builder::transaction_builder DEBUG queue:
2024-03-13 19:32:14 (UTC) iota_sdk::client::api::block_builder::transaction_builder DEBUG queue 2
2024-03-13 19:32:14 (UTC) iota_sdk::client::api::block_builder::transaction_builder DEBUG queue:
2024-03-13 19:32:14 (UTC) iota_sdk::client::api::block_builder::transaction_builder DEBUG queue 2
2024-03-13 19:32:14 (UTC) iota_sdk::client::api::block_builder::transaction_builder DEBUG queue:
2024-03-13 19:32:14 (UTC) iota_sdk::client::api::block_builder::transaction_builder::requirement::account DEBUG AccountId(0x3333333333333333333333333333333333333333333333333333333333333333) requirement already fulfilled by OutputId { id: "0xe21a070f8a2d685ea29a701f171271cce9823f3d83e00ccc9327e96ec7e63eb05a0000000000", transaction_id: TransactionId { id: "0xe21a070f8a2d685ea29a701f171271cce9823f3d83e00ccc9327e96ec7e63eb05a000000", slot_index: SlotIndex(90) }, output_index: 0 }
2024-03-13 19:32:14 (UTC) iota_sdk::client::api::block_builder::transaction_builder::requirement DEBUG Fulfilling requirement Account(AccountId(0x1111111111111111111111111111111111111111111111111111111111111111))
2024-03-13 19:32:14 (UTC) iota_sdk::client::api::block_builder::transaction_builder DEBUG queue:
2024-03-13 19:32:14 (UTC) iota_sdk::client::api::block_builder::transaction_builder DEBUG queue 2
2024-03-13 19:32:14 (UTC) iota_sdk::client::api::block_builder::transaction_builder DEBUG queue:
2024-03-13 19:32:14 (UTC) iota_sdk::client::api::block_builder::transaction_builder DEBUG queue 2
2024-03-13 19:32:14 (UTC) iota_sdk::client::api::block_builder::transaction_builder::requirement::account DEBUG AccountId(0x1111111111111111111111111111111111111111111111111111111111111111) requirement already fulfilled by OutputId { id: "0x1f0cdc1bf552449906ea7139f6a29ddd0e7619915841b6de8cf73032226df55f5a0000000000", transaction_id: TransactionId { id: "0x1f0cdc1bf552449906ea7139f6a29ddd0e7619915841b6de8cf73032226df55f5a000000", slot_index: SlotIndex(90) }, output_index: 0 }
2024-03-13 19:32:14 (UTC) iota_sdk::client::api::block_builder::transaction_builder::requirement DEBUG Fulfilling requirement Account(AccountId(0x2222222222222222222222222222222222222222222222222222222222222222))
2024-03-13 19:32:14 (UTC) iota_sdk::client::api::block_builder::transaction_builder DEBUG queue:
2024-03-13 19:32:14 (UTC) iota_sdk::client::api::block_builder::transaction_builder DEBUG queue 2
2024-03-13 19:32:14 (UTC) iota_sdk::client::api::block_builder::transaction_builder::requirement::account DEBUG AccountId(0x2222222222222222222222222222222222222222222222222222222222222222) requirement already fulfilled by OutputId { id: "0x8d144dd04605a0d249abed652c320b580f96c7119d1296bb600eadb1cd1160155a0000000000", transaction_id: TransactionId { id: "0x8d144dd04605a0d249abed652c320b580f96c7119d1296bb600eadb1cd1160155a000000", slot_index: SlotIndex(90) }, output_index: 0 }
2024-03-13 19:32:14 (UTC) iota_sdk::client::api::block_builder::transaction_builder::requirement DEBUG Fulfilling requirement NativeTokens
2024-03-13 19:32:14 (UTC) iota_sdk::client::api::block_builder::transaction_builder DEBUG queue:
2024-03-13 19:32:14 (UTC) iota_sdk::client::api::block_builder::transaction_builder DEBUG queue 2
2024-03-13 19:32:14 (UTC) iota_sdk::client::api::block_builder::transaction_builder DEBUG queue:
2024-03-13 19:32:14 (UTC) iota_sdk::client::api::block_builder::transaction_builder DEBUG queue 2
2024-03-13 19:32:14 (UTC) iota_sdk::client::api::block_builder::transaction_builder DEBUG queue:
2024-03-13 19:32:14 (UTC) iota_sdk::client::api::block_builder::transaction_builder DEBUG queue:
2024-03-13 19:32:14 (UTC) iota_sdk::client::api::block_builder::transaction_builder::requirement::native_tokens DEBUG Native tokens requirement already fulfilled
2024-03-13 19:32:14 (UTC) iota_sdk::client::api::block_builder::transaction_builder::requirement DEBUG Fulfilling requirement Amount
2024-03-13 19:32:14 (UTC) iota_sdk::client::api::block_builder::transaction_builder DEBUG queue:
2024-03-13 19:32:14 (UTC) iota_sdk::client::api::block_builder::transaction_builder DEBUG queue 2
2024-03-13 19:32:14 (UTC) iota_sdk::client::api::block_builder::transaction_builder DEBUG queue:
2024-03-13 19:32:14 (UTC) iota_sdk::client::api::block_builder::transaction_builder DEBUG queue 2
2024-03-13 19:32:14 (UTC) iota_sdk::client::api::block_builder::transaction_builder DEBUG queue:
2024-03-13 19:32:14 (UTC) iota_sdk::client::api::block_builder::transaction_builder DEBUG queue:
2024-03-13 19:32:14 (UTC) iota_sdk::client::api::block_builder::transaction_builder DEBUG queue:
2024-03-13 19:32:14 (UTC) iota_sdk::client::api::block_builder::transaction_builder DEBUG queue 2
2024-03-13 19:32:14 (UTC) iota_sdk::client::api::block_builder::transaction_builder DEBUG queue:
2024-03-13 19:32:14 (UTC) iota_sdk::client::api::block_builder::transaction_builder DEBUG queue 2
2024-03-13 19:32:14 (UTC) iota_sdk::client::api::block_builder::transaction_builder DEBUG queue:
2024-03-13 19:32:14 (UTC) iota_sdk::client::api::block_builder::transaction_builder DEBUG queue:
2024-03-13 19:32:14 (UTC) iota_sdk::client::api::block_builder::transaction_builder DEBUG queue:
2024-03-13 19:32:14 (UTC) iota_sdk::client::api::block_builder::transaction_builder DEBUG queue 2
2024-03-13 19:32:14 (UTC) iota_sdk::client::api::block_builder::transaction_builder DEBUG queue:
2024-03-13 19:32:14 (UTC) iota_sdk::client::api::block_builder::transaction_builder DEBUG queue 2
2024-03-13 19:32:14 (UTC) iota_sdk::client::api::block_builder::transaction_builder DEBUG queue:
2024-03-13 19:32:14 (UTC) iota_sdk::client::api::block_builder::transaction_builder DEBUG queue:
2024-03-13 19:32:14 (UTC) iota_sdk::client::api::block_builder::transaction_builder DEBUG queue:
2024-03-13 19:32:14 (UTC) iota_sdk::client::api::block_builder::transaction_builder DEBUG queue 2
2024-03-13 19:32:14 (UTC) iota_sdk::client::api::block_builder::transaction_builder::requirement::amount DEBUG Amount requirement already fulfilled
2024-03-13 19:32:14 (UTC) iota_sdk::client::api::block_builder::transaction_builder::requirement DEBUG Fulfilling requirement Mana
2024-03-13 19:32:14 (UTC) iota_sdk::client::api::block_builder::transaction_builder DEBUG queue:
2024-03-13 19:32:14 (UTC) iota_sdk::client::api::block_builder::transaction_builder DEBUG queue 2
2024-03-13 19:32:14 (UTC) iota_sdk::client::api::block_builder::transaction_builder DEBUG queue:
2024-03-13 19:32:14 (UTC) iota_sdk::client::api::block_builder::transaction_builder DEBUG queue 2
2024-03-13 19:32:14 (UTC) iota_sdk::client::api::block_builder::transaction_builder DEBUG queue:
2024-03-13 19:32:14 (UTC) iota_sdk::client::api::block_builder::transaction_builder DEBUG queue:
2024-03-13 19:32:14 (UTC) iota_sdk::client::api::block_builder::transaction_builder::requirement::mana DEBUG Mana requirement selected mana: 0, required mana: 0
2024-03-13 19:32:14 (UTC) iota_sdk::client::api::block_builder::transaction_builder::requirement::mana DEBUG Mana requirement already fulfilled
2024-03-13 19:32:14 (UTC) iota_sdk::client::api::block_builder::transaction_builder DEBUG queue:
2024-03-13 19:32:14 (UTC) iota_sdk::client::api::block_builder::transaction_builder DEBUG queue 2
2024-03-13 19:32:14 (UTC) iota_sdk::client::api::block_builder::transaction_builder DEBUG queue:
2024-03-13 19:32:14 (UTC) iota_sdk::client::api::block_builder::transaction_builder DEBUG queue 2
2024-03-13 19:32:14 (UTC) iota_sdk::client::api::block_builder::transaction_builder DEBUG queue:
2024-03-13 19:32:14 (UTC) iota_sdk::client::api::block_builder::transaction_builder DEBUG queue:
2024-03-13 19:32:14 (UTC) iota_sdk::client::api::block_builder::transaction_builder DEBUG queue:
2024-03-13 19:32:14 (UTC) iota_sdk::client::api::block_builder::transaction_builder DEBUG queue 2
2024-03-13 19:32:14 (UTC) iota_sdk::client::api::block_builder::transaction_builder DEBUG queue:
2024-03-13 19:32:14 (UTC) iota_sdk::client::api::block_builder::transaction_builder DEBUG queue 2
2024-03-13 19:32:14 (UTC) iota_sdk::client::api::block_builder::transaction_builder DEBUG queue:
2024-03-13 19:32:14 (UTC) iota_sdk::client::api::block_builder::transaction_builder DEBUG queue:
2024-03-13 19:32:14 (UTC) iota_sdk::client::api::block_builder::transaction_builder DEBUG queue:
2024-03-13 19:32:14 (UTC) iota_sdk::client::api::block_builder::transaction_builder DEBUG queue 2
2024-03-13 19:32:14 (UTC) iota_sdk::client::api::block_builder::transaction_builder DEBUG queue:
2024-03-13 19:32:14 (UTC) iota_sdk::client::api::block_builder::transaction_builder DEBUG queue 2
2024-03-13 19:32:14 (UTC) iota_sdk::client::api::block_builder::transaction_builder DEBUG queue:
2024-03-13 19:32:14 (UTC) iota_sdk::client::api::block_builder::transaction_builder DEBUG queue:
2024-03-13 19:32:14 (UTC) iota_sdk::client::api::block_builder::transaction_builder::remainder DEBUG input_nts: {}
2024-03-13 19:32:14 (UTC) iota_sdk::client::api::block_builder::transaction_builder::remainder DEBUG output_nts: {}
2024-03-13 19:32:14 (UTC) iota_sdk::client::api::block_builder::transaction_builder DEBUG queue:
2024-03-13 19:32:14 (UTC) iota_sdk::client::api::block_builder::transaction_builder DEBUG queue 2
2024-03-13 19:32:14 (UTC) iota_sdk::client::api::block_builder::transaction_builder DEBUG queue:
2024-03-13 19:32:14 (UTC) iota_sdk::client::api::block_builder::transaction_builder DEBUG queue 2
2024-03-13 19:32:14 (UTC) iota_sdk::client::api::block_builder::transaction_builder DEBUG queue:
2024-03-13 19:32:14 (UTC) iota_sdk::client::api::block_builder::transaction_builder DEBUG queue:
2024-03-13 19:32:14 (UTC) iota_sdk::client::api::block_builder::transaction_builder DEBUG queue:
2024-03-13 19:32:14 (UTC) iota_sdk::client::api::block_builder::transaction_builder DEBUG queue 2
2024-03-13 19:32:14 (UTC) iota_sdk::client::api::block_builder::transaction_builder::remainder DEBUG No remainder required
2024-03-13 19:32:14 (UTC) iota_sdk::client::api::block_builder::transaction_builder DEBUG queue:
2024-03-13 19:32:14 (UTC) iota_sdk::client::api::block_builder::transaction_builder DEBUG queue 2
2024-03-13 19:32:14 (UTC) iota_sdk::client::api::block_builder::transaction_builder DEBUG queue:
2024-03-13 19:32:14 (UTC) iota_sdk::client::api::block_builder::transaction_builder DEBUG queue 2
2024-03-13 19:32:14 (UTC) iota_sdk::client::api::block_builder::transaction_builder DEBUG queue:
2024-03-13 19:32:14 (UTC) iota_sdk::client::api::block_builder::transaction_builder DEBUG queue:
2024-03-13 19:32:14 (UTC) iota_sdk::client::api::block_builder::transaction_builder DEBUG queue:
2024-03-13 19:32:14 (UTC) iota_sdk::client::api::block_builder::transaction_builder DEBUG queue 2
2024-03-13 19:32:14 (UTC) iota_sdk::client::api::block_builder::transaction_builder DEBUG queue:
2024-03-13 19:32:14 (UTC) iota_sdk::client::api::block_builder::transaction_builder DEBUG queue 2
2024-03-13 19:32:14 (UTC) iota_sdk::client::api::block_builder::transaction_builder DEBUG queue:
2024-03-13 19:32:14 (UTC) iota_sdk::client::api::block_builder::transaction_builder DEBUG queue:

```
That's only for 3 inputs, seem like quite a lot of action, isn't that worse than ordering once at the end?

Reply:
You have a point. Perhaps we can iterate unsorted for times when it doesn't matter

Reply:
I separated the sorted and unsorted iteration so now it will only do the sorting for places that need it